client/packages/programs/src/JsonForms/common/components/Array/FirstItemArray.tsx

```diff
@@ -63,6 +67,12 @@ const FirstItemArrayComponent = (props: FirstItemArrayControlCustomProps) => {
     [uischemas, schema, uischema.scope, path, uischema, rootSchema]
   );
 
+  // If there are definitions they need to be added to the child schema
+  const schemaWithDefs = {
+    ...(schema as JsonSchema7),
+    definitions: rootSchema.definitions as Record<string, JsonSchema7>,
+  };
```

Comment:
This seems slightly hacky but seems to work...

Reply:
:nit I would have the `rootSchema.definitions` first and spread the schema, so that the schema takes precedence if there are any conflicts. Realistically though, I don't think that it makes any difference in this case.